cluster-autoscaler/metrics/liveness.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package metrics
+
+import (
+	"fmt"
+	"net/http"
+	"sync"
+	"time"
+)
+
+// HealthCheck contains information about last time of autoscaler activity and timeout
+type HealthCheck struct {
+	lastActivity time.Time
+	mutex        *sync.Mutex
+	timeout      time.Duration
+	checkTimeout bool
+}
+
+// NewHealthCheck builds new HealthCheck object with given timeout
+func NewHealthCheck(timeout time.Duration) *HealthCheck {
+	return &HealthCheck{
+		lastActivity: time.Now(),
+		mutex:        &sync.Mutex{},
+		timeout:      timeout,
+		checkTimeout: false,
+	}
+}
+
+// StartMonitoring activates checks for autoscaler inactivity
+func (hc *HealthCheck) StartMonitoring() {
+	hc.mutex.Lock()
+	defer hc.mutex.Unlock()
+	hc.checkTimeout = true
```

Comment:
I'm thinking we should also update lastActivity here. Otherwise we risk failing a probe between StartMonitoring() and first UpdateLastActivity(). There is 10s delay between those 2 calls in main.go.